vault/identity_store_oidc_provider.go

```diff
@@ -277,7 +278,163 @@ func oidcProviderPaths(i *IdentityStore) []*framework.Path {
 			HelpSynopsis:    "Query OIDC configurations",
 			HelpDescription: "Query this path to retrieve the configured OIDC Issuer and Keys endpoints, response types, subject types, and signing algorithms used by the OIDC backend.",
 		},
+		{
+			Pattern: "oidc/provider/" + framework.GenericNameRegex("name") + "/.well-known/keys",
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeString,
+					Description: "Name of the provider",
+				},
+			},
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ReadOperation: i.pathOIDCReadProviderPublicKeys,
+			},
+			HelpSynopsis:    "Retrieve public keys",
+			HelpDescription: "Returns the public portion of keys for a named OIDC provider. Clients can use them to validate the authenticity of an ID token.",
+		},
+	}
+}
+
+// TODO: load clients into memory (go-memdb) to look this up
+func (i *IdentityStore) allClients(ctx context.Context, s logical.Storage) (*[]client, error) {
+	clientNames, err := s.List(ctx, clientPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var clients []client
+	for _, name := range clientNames {
+		entry, err := s.Get(ctx, clientPath+name)
```

Comment:
This bit can incur a heavy storage cost as the number of clients grow.

Reply:
Yes, definitely. Austin and I discussed that we will need to document the performance implications of setting `allowed_client_ids="*"`. I am happy to consider other options for this case but at the moment I don't see another way around it.